src/anet.c

```diff
@@ -680,3 +682,19 @@ int anetPipe(int fds[2], int read_flags, int write_flags) {
     close(fds[1]);
     return -1;
 }
+
+int anetSockId(char *err, int fd, uint32_t id)
+{
+#ifdef HAVE_SOCKOPTID
+        if (setsockopt(fd, SOL_SOCKET, SOCKOPTID, (void *)&id, sizeof(id)) == -1) {
+            anetSetError(err, "setsockopt: %s", strerror(errno));
+	    return ANET_ERR;
+        }
+	return ANET_OK;
+#else
+        UNUSED(fd);
+        if (id > 0)
+            anetSetError(err,"anetSockid unsupported on this platform");
+        return ANET_OK;
```

Comment:
are we sure we wanna return OK here?
alternatively, maybe we should add an `isValid` callback to the config, and fail the configuration when this is not supported?